pygmt/clib/session.py

```diff
@@ -1838,19 +1847,19 @@ def virtualfile_in(  # noqa: PLR0912
             )
             mincols = 3
 
-        kind = data_kind(data, required=required_data)
+        kind = data_kind(data, required=required)
         _validate_data_input(
             data=data,
             x=x,
             y=y,
             z=z,
             mincols=mincols,
-            required_data=required_data,
+            required=required,
             kind=kind,
         )
 
         if check_kind:
-            valid_kinds = ("file", "arg") if required_data is False else ("file",)
+            valid_kinds = ("file", "arg") if required is False else ("file",)
```

Comment:
I feel that `required` by itself isn't a very descriptive parmeter name. This `required_data` parameter was added in #2493 to better handle optional virtual files (via the 'arg' kind). Would a name like `accept_vfile` (default to False) or something along those lines work? Then the check would be:

```python
            valid_kinds = ("file", "arg") if accept_vfile else ("file",)
```

Or we could also just keep the `required_data` name as is? Thoughts?

Reply:
> Would a name like `optional_vfile` (default to True) or something along those lines work?

You meant `optional_vfile` default to False?

Reply:
Ah yes, I meant `optional_vfile=False` by default, but that's probably not a great name either. On second thought, let's go with the `required` parameter name then. Using just `required` in `data_kind` makes sense because we're only passing in an argument to `data`, and I guess we do validate whether `data`/`x`/`y` are not None in `virtualfile_in` so there shouldn't be any ambiguity, but need to update the docstring a bit.